container/crio/handler.go

```diff
@@ -305,11 +305,18 @@ func (h *crioContainerHandler) GetStats() (*info.ContainerStats, error) {
 	if err != nil {
 		return stats, err
 	}
-	// Clean up stats for containers that don't have their own network - this
-	// includes containers running in Kubernetes pods that use the network of the
-	// infrastructure container. This stops metrics being reported multiple times
-	// for each container in a pod.
-	if !h.needNet() {
+
+	if h.needNet() && len(stats.Network.Interfaces) == 0 {
+		// No network related information indicates that the pid of the
+		// container is not longer valid and we need to ask crio to
+		// provide the pid of another container from that pod
+		h.pidKnown = false
+		return stats, nil
```

Comment:
I would prefer these two conditions are switched so we don't have to call needsNet twice:

```
if !h.needNet() {
....
} else if len(...) {
....
}
```

Reply:
@haircommander Done. Thanks.